File: services/products/src/main/java/br/com/deltaglobalbank/products/features/lending/createLendingProduct/CreateLendingProductRequest.java
package br.com.deltaglobalbank.products.features.lending.createLendingProduct;

import java.math.BigDecimal;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;

public record CreateLendingProductRequest(
    @NotBlank String agreementName,
    String displayName,
    @NotNull @PositiveOrZero BigDecimal minMonthlyRate,
    @NotNull @PositiveOrZero BigDecimal maxMonthlyRate,
    @Positive int minMonths,
    @Positive int maxMonths,
    @NotNull @PositiveOrZero BigDecimal minAmount,
    @NotNull @PositiveOrZero BigDecimal maxAmount,
    BigDecimal commissionRate
) {
}

File: services/products/src/main/kotlin/br/com/deltaglobalbank/products/features/lending/createLendingProduct/CreateLendingProductRequest.kt
package br.com.deltaglobalbank.products.features.lending.createLendingProduct

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

data class CreateLendingProductRequest(
    @field:NotBlank
    val agreementName: String,

    val displayName: String? = null,

    @field:NotNull @field:PositiveOrZero
    val minMonthlyRate: BigDecimal,

    @field:NotNull @field:PositiveOrZero
    val maxMonthlyRate: BigDecimal,

    @field:Positive
    val minMonths: Int,

    @field:Positive
    val maxMonths: Int,

    @field:NotNull @field:PositiveOrZero
    val minAmount: BigDecimal,

    @field:NotNull @field:PositiveOrZero
    val maxAmount: BigDecimal,

    val commissionRate: BigDecimal? = null,
)